### PR TITLE
Run golang-ci linter in travisci. Add golang-ci configuration

### PR DESCRIPTION
**Pull request by Dean-Coakley**
### Proposed changes
* Add `golangci-lint` config to set what linters to run
* Add linting step to travis.yml so build will fail if format is bad.

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [x] I have rebased my branch onto master
- [x] I will ensure my PR is targeting the master branch and pulling from my branch from my own fork

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -19,3 +19,5 @@ before_install:
   wget https://${fossalocation}/${fossafile} && tar xzf ${fossafile}
   && ./fossa init
   && GO111MODULE=on FOSSA_API_KEY=${fossapush} ./fossa analyze -t kubernetes-ingress -b ${TRAVIS_BRANCH}; fi
+- echo "Checking ingress controller for linting errors" && wget -O - -q ${golangci_lint} | sh -s v1.17.1
```
Comment (pleshakov) [inline .travis.yml]: `sh -s v1.17.1` ? 
judging by the output from here https://travis-ci.org/nginxinc/kubernetes-ingress/builds/575403783?utm_source=github_status&utm_medium=notification, `$ export golangci_lint_version=v1.17.1`,  the env variable was supposed to be used here

File: .golangci.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    enable:
+    - golint
+    - goimports
+    - misspell
+    - gofmt
+    - unparam
+    - unconvert
+    - interfacer
+    disable:
+    - maligned
```
Comment (pleshakov) [inline .golangci.yml]: aren't those disabled by default https://github.com/golangci/golangci-lint#disabled-by-default-linters--e--enable ?

File: .golangci.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+linters-settings:
+  misspell:
+    locale: US
+
+linters:
+    enable:
```
Comment (pleshakov) [inline .golangci.yml]: all those linters are enabled by default and there are more enabled linters!
if you want to control which are enabled, consider the following example from the docs https://github.com/golangci/golangci-lint#config-file:
```
linters:
  enable:
    - megacheck
    - govet
  enable-all: false
  disable:
    - maligned
    - prealloc
  disable-all: false
  presets:
    - bugs
    - unused
  fast: false
```
I think we should lock down the linters that way

File: .travis.yml [hunk 4]
```diff
@@ -19,3 +19,5 @@ before_install:
...
+- ./bin/golangci-lint run --config .golangci.yml
```
Comment (pleshakov) [inline .travis.yml]: note that .golangci.yml will be used even if you don't specify `--config`

Comment (pleshakov) [reply .travis.yml]: also, we have `make lint` target, can we use that?

File: .travis.yml (hunk 1 above)
Comment (Dean-Coakley) [reply .travis.yml]: I decided against using a travisci variable as you saw in that log there. 
Version of the linter should be tracked in version control.

File: .golangci.yml (hunk 2 above)
Comment (Dean-Coakley) [reply .golangci.yml]: I found the following to not be enabled by default.
```
gofmt: Gofmt checks whether code was gofmt-ed. By default this tool runs with -s option to check for code 
goimports: Goimports does everything that gofmt does. Additionally it checks unused imports [fast: true, auto-fix: true]
interfacer: Linter that suggests narrower interface types [fast: false, auto-fix: false]
misspell: Finds commonly misspelled English words in comments [fast: true, auto-fix: true]
unconvert: Remove unnecessary type conversions [fast: true, auto-fix: false]
unparam: Reports unused function parameters [fast: false, auto-fix: false]
```

Tested with `$ golangci-lint linters` in dir with config and outside it.

File: .golangci.yml (hunk 3 above)
Comment (Dean-Coakley) [reply .golangci.yml]: Locking down the linters like that looks nice though, I'll investigate that.

**Pull request merged**